Add neofetch@eson57

eson57 opened the pull request:
Basic system info on top of your right-click fingertip.

rcalixte commented on neofetch@eson57/files/neofetch@eson57/neofetch@eson57.sh:
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+gnome-terminal -- neofetch
```
There is a way to retrieve the logged in user's default terminal and the exec arguments to include. There are two gsettings (`exec` and `exec-arg`) under org.cinnamon.desktop.default-applications.terminal (/org/cinnamon/desktop/applications/terminal if you browse with dconf) that contain these values.

So you can retrieve the two values like:
```bash
EXEC=$(gsettings get org.cinnamon.desktop.default-applications.terminal exec)
EXECARG=$(gsettings get org.cinnamon.desktop.default-applications.terminal exec-arg)
```

I realize this is the same structure that was used in the run-speedtest Action so I apologize for the oversight there. Also, I tested both run-speedtest and this and neither were able to work for me without modification. More on that in the second comment.

eson57 replied:
Last thing first! In gnome-terminal there is this setting under `Edit/Preferences/<your profile>/Command/When command exits`, that have to be set to 'Hold the terminal open'. I think default is 'Exit the terminal' and I haven't found a way to overrule this. The setting obviously overrules the `$SHELL` string.

eson57 replied:
The rest what you suggest is a bit over my level of knowledge.
However, I found the below to identify default terminal and that seem to work, but I fail getting the script to run the neofetch command. So I need help with any of these solutions.

```
# Detect the default terminal emulator
TERMINAL_CMD=""
if command -v x-terminal-emulator &>/dev/null; then
    TERMINAL_CMD="x-terminal-emulator"
elif command -v gnome-terminal &>/dev/null; then
    TERMINAL_CMD="gnome-terminal"
elif command -v konsole &>/dev/null; then
    TERMINAL_CMD="konsole"
else
    echo "Error: No supported terminal emulator found."
    exit 1
fi
# Open a new terminal window and execute 'neofetch'
$TERMINAL_CMD -- neofetch
```

rcalixte replied:
> Last thing first! In gnome-terminal there is this setting under `Edit/Preferences/<your profile>/Command/When command exits`, that have to be set to 'Hold the terminal open'. I think default is 'Exit the terminal' and I haven't found a way to overrule this. The setting obviously overrules the `$SHELL` string.

Don't change this. People have their terminals configured for a reason. And again, this does nothing for the people who don't have the terminal installed or are using another as their preferred application.

rcalixte replied:
> The rest what you suggest is a bit over my level of knowledge. However, I found the below to identify default terminal and that seem to work, but I fail getting the script to run the neofetch command. So I need help with any of these solutions.
> 
> ```
> #!/usr/bin/bash
> # Detect the default terminal emulator
> TERMINAL_CMD=""
> if command -v x-terminal-emulator &>/dev/null; then
>     TERMINAL_CMD="x-terminal-emulator"
> elif command -v gnome-terminal &>/dev/null; then
>     TERMINAL_CMD="gnome-terminal"
> elif command -v konsole &>/dev/null; then
>     TERMINAL_CMD="konsole"
> else
>     echo "Error: No supported terminal emulator found."
>     exit 1
> fi
> # Open a new terminal window and execute 'neofetch'
> $TERMINAL_CMD -- neofetch
> ```

You can copy and paste the exact two lines I provided to fetch the values regardless of what the configuration is. The code that you've proposed here fails too many conditions otherwise.

rcalixte replied:
> The setting obviously overrules the `$SHELL` string.

This is the key thing. Maybe we should just skip these kinds of Actions for now. It's easy enough for most to open a terminal and type a single command just as easy as it is to install an Action. We could save some time and energy that would be better spent elsewhere.

eson57 replied:
Sure, I can still use it as is internally.

rcalixte commented on neofetch@eson57/neofetch@eson57.nemo_action.in:
```diff
@@ -0,0 +1,8 @@
+[Nemo Action]
+_Name=Neofetch
+Exec=<neofetch@eson57/neofetch@eson57.sh %F>
+Icon-Name=dialog-information-symbolic
+Selection=none
+Extensions=any;
+Conditions=dbus org.Cinnamon;desktop;
+Dependencies=gnome-terminal;neofetch;
```
Following up on the first comment, you can remove the dependency for the terminal by just retrieving whatever the user has configured as their default. But the Exec line shouldn't have the `%F` parameter since you have `desktop` in the Conditions. (I can modify the validation script to catch this in the future.) This also affects run-speedtest if you can amend that too.

eson57 replied:
Will be done.

rcalixte commented:
How do you want to proceed with the run-speedtest Action? Should we remove that?

eson57 commented:
You're the boss. You decide.

onsdag den 6 mars 2024 skrev Rick Calixte:
> How do you want to proceed with the run-speedtest Action? Should we remove that?
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/linuxmint/cinnamon-spices-actions/pull/261#issuecomment-1981047773
> You are receiving this because you modified the open/close state.
> 
> Message ID: ***@***.***

-- 
Sent from SailfishX

rcalixte commented:
> You're the boss. You decide.

I wish. 😄 

My leaning is that if it isn't working for most users and won't be fixed, then we should remove it.

eson57 commented:
Then remove it. I don't care that much. I was just amusing myself by trying to do something substantial. I learn by doing.
But, of course, I wonder if people are really using that old default setting. I changed it probably 10 years ago. But, of course, I wonder if people are really using that old default setting. I changed it probably 10 years ago. I'd be mad if the terminal closed as soon as I ran a command.

rcalixte commented:
> I wonder if people are really using that old default setting.

It's the default and many users do not change defaults. You have to keep in mind the number of users who also have done new installations rather than carrying forward via upgrades as well. I have all of my local terminals configured to exit as well. That was why I wasn't able to get either of these to work. It's important to understand that it's a preference for a reason. It's also important to try to make a tool or application as useful for as many people as possible and that often starts by assuming default settings. I've encountered this issue as a challenge myself so it's not an easy thing to always keep top of mind. That being said, that challenge is what keeps things interesting sometimes. 😅



> Then remove it.

I'll go ahead and get that done for you. If you find a working solution, please don't hesitate to share with the broader community!
